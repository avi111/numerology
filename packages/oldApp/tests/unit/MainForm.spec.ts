import {mount, Wrapper} from '@vue/test-utils';
import MainForm from '@/components/MainForm.vue';
import sinon from 'sinon';

let wrapper: Wrapper<MainForm>;

describe('MainForm mount regularly', () => {
    beforeEach(() => {
        wrapper = mount(MainForm, {
            computed: {
                dictionary() {
                    return [];
                },
            },
        });
    });
    it.skip('matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('form have all inputs', () => {
        const form = wrapper.find('form');
        expect(form.exists()).toBeTruthy();
        const formData = Array.from(new FormData(form.element as HTMLFormElement).entries());
        const fields = formData.map((el) => el[0]);
        const expectedFields = Object.keys(wrapper.vm.$data.inputProps).length;
        expect(fields.length).toBe(expectedFields);
    });

    it('resets on reset', () => {
        const firstName = 'אבי';
        wrapper.setData({
            inputProps: {
                firstName,
            },
        });

        expect(wrapper.vm.$data.inputProps.firstName).toBe(firstName);

        const form = wrapper.find('form');
        form.trigger('reset');
        expect(wrapper.vm.$data.inputProps.firstName).toBe('');
    });

    // skipped due to cancellation of html5 pattern in favour of validateOnlyHebrew
    it.skip('prevents english chars', () => {
        const english = 'avi';
        const input = wrapper.find('[name="firstName"]');
        input.setValue(english);
        expect((input.element as HTMLInputElement).checkValidity()).toBeFalsy();
        expect(input.classes().indexOf('is-invalid')).not.toBe(-1);
    });

    // skipped due to cancellation of html5 pattern in favour of validateOnlyHebrew
    it.skip('prevents english chars from mixed name', () => {
        const hebrew = 'אבי g';
        const input = wrapper.find('[name="firstName"]');
        input.setValue(hebrew);
        expect((input.element as HTMLInputElement).checkValidity()).toBeFalsy();
        expect(input.classes().indexOf('is-invalid')).not.toBe(-1);
    });

    // skipped due to cancellation of html5 pattern in favour of validateOnlyHebrew
    it.skip('accepts hebrew chars', () => {
        const hebrew = 'אבי';
        const input = wrapper.find('[name="firstName"]');
        input.setValue(hebrew);
        expect((input.element as HTMLInputElement).checkValidity()).toBeTruthy();
        expect(input.classes().indexOf('is-invalid')).toBe(-1);
    });
});

describe('check functionality of MainForm', () => {
    it('triggers onSubmit on submit', () => {
        const onSubmit = sinon.spy();
        wrapper = mount(MainForm, {
            methods: {
                onSubmit,
            },
            computed: {
                dictionary() {
                    return [];
                },
            },
        });
        const form = wrapper.find('form');
        form.trigger('submit');
        expect(onSubmit.called).toBeTruthy();
    });

    it('triggers onReset on reset', () => {
        const spy = sinon.spy();
        wrapper = mount(MainForm, {
            methods: {
                onReset: spy,
            },
            computed: {
                dictionary() {
                    return [];
                },
            },
        });
        const form = wrapper.find('form');
        form.trigger('reset');
        expect(spy.called).toBeTruthy();
    });
});
