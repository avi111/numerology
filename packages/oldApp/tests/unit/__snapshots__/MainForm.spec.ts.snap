// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainForm mount regularly matches snapshot 1`] = `
<div class="hello">
  <form id="form" class="">
    <div class="form-row">
      <div class="col-md-6">
        <fieldset id="patient-details" class="form-group">
          <!---->
          <div tabindex="-1" role="group">
            <div role="group" class="input-group">
              <!----><input name="firstName" type="text" placeholder="" required="required" aria-required="true" aria-invalid="true" class="form-control is-invalid" pattern="[א-ת\\s]+" id="__BVID__5">
              <div class="input-group-append"><button type="button" class="btn btn-info">*</button></div>
              <!---->
            </div>
            <div role="group" class="input-group">
              <!----><input name="firstNameAtBirth" type="text" placeholder="()" aria-invalid="true" class="form-control is-invalid" pattern="[א-ת\\s]+" id="__BVID__6">
              <div class="input-group-append"><button id="tooltip-firstNameAtBirth" type="button" class="btn btn-info">?
                </button>
                <!---->
              </div>
              <!---->
            </div>
            <div role="group" class="input-group">
              <!----><input name="familyName" type="text" placeholder="" required="required" aria-required="true" aria-invalid="true" class="form-control is-invalid" pattern="[א-ת\\s]+" id="__BVID__8">
              <div class="input-group-append"><button id="tooltip-familyName" type="button" class="btn btn-info">?</button>
                <!---->
              </div>
              <!---->
            </div>
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
        <fieldset class="form-group" id="__BVID__10">
          <!---->
          <div tabindex="-1" role="group">
            <div class="form-row">
              <div class="col-md-6">
                <div role="group" class="input-group">
                  <!---->
                  <div class="vdatetime"> <input id="birthDate" type="text" name="birthDate" placeholder="" reqired="" class="vdatetime-input">
                    <!----> <span name="vdatetime-fade" tag="div"></span></div>
                  <div class="input-group-append"><button type="button" class="btn btn-info">*</button></div>
                  <!---->
                </div>
              </div>
              <div class="col-md-6">
                <div role="group" class="input-group">
                  <!---->
                  <div class="vdatetime"> <input id="birthHour" type="text" name="birthHour" placeholder="" class="vdatetime-input">
                    <!----> <span name="vdatetime-fade" tag="div"></span></div>
                  <div class="input-group-append"><button id="tooltip-birthHour" type="button" class="btn btn-info">?</button>
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
        <fieldset class="form-group" id="__BVID__16">
          <!---->
          <div tabindex="-1" role="group"><select name="gender" required="required" aria-required="true" class="custom-select" id="__BVID__17">
              <option value="male"></option>
              <option value="female"></option>
            </select>
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
      </div>
      <div class="col-md-6">
        <fieldset id="patient-parents" class="form-group">
          <!---->
          <div tabindex="-1" role="group">
            <div role="group" class="input-group">
              <!----><input name="fatherName" type="text" placeholder="" required="required" aria-required="true" aria-invalid="true" class="form-control is-invalid" pattern="[א-ת\\s]+" id="__BVID__19">
              <div class="input-group-append">
                <div class="input-group-append"><button type="button" class="btn btn-info">*</button></div>
              </div>
              <!---->
            </div>
            <div role="group" class="input-group">
              <!----><input name="fatherNameAtBirthOfPatient" type="text" placeholder=" ( )" aria-invalid="true" class="form-control is-invalid" id="__BVID__20">
              <div class="input-group-append"><button id="tooltip-fatherNameAtBirthOfPatient" type="button" class="btn btn-info">?
                </button>
                <!---->
              </div>
              <!---->
            </div>
            <div role="group" class="input-group">
              <!----><input name="motherName" type="text" placeholder="" required="required" aria-required="true" aria-invalid="true" class="form-control is-invalid" pattern="[א-ת\\s]+" id="__BVID__22">
              <div class="input-group-append">
                <div class="input-group-append"><button type="button" class="btn btn-info">*</button></div>
              </div>
              <!---->
            </div>
            <div role="group" class="input-group">
              <!----><input name="motherNameAtBirthOfPatient" type="text" placeholder=" ( )" aria-invalid="true" class="form-control is-invalid" pattern="[א-ת\\s]+" id="__BVID__23">
              <div class="input-group-append"><button id="tooltip-motherNameAtBirthOfPatient" type="button" class="btn btn-info">?
                </button>
                <!---->
              </div>
              <!---->
            </div>
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
      </div>
    </div>
    <div class="form-row">
      <div class="col-md-12">
        <fieldset class="form-group" id="__BVID__25">
          <!---->
          <div tabindex="-1" role="group"><button type="submit" class="btn btn-primary">
            </button> <button type="reset" class="btn btn-danger">
            </button>
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
      </div>
    </div>
  </form>
</div>
`;
